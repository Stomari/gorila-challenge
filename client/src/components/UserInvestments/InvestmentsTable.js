import React from 'react';

const InvestmentsTable = ({ investmentInfo, deleteInvestment }) => {
  return (
    <table>
      <thead>
        <tr>
          <th>Data da compra</th>
          <th>Valor</th>
        </tr>
      </thead>
      <tbody>
        {
          investmentInfo && investmentInfo.map((element) => {
            return (
              <tr key={element._id}>
                <td>{element.date}</td>
                <td>{element.value}</td>
                <td><img className="delete" onClick={() => deleteInvestment(element._id)} src="./images/trash.png" alt="Trash Icon" /></td>
              </tr>
            )
          })
        }
      </tbody>
    </table>
  )
}

export default InvestmentsTable;