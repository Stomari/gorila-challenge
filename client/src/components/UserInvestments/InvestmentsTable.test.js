import React from 'react';
import { shallow } from 'enzyme';
import InvestmentsTable from './InvestmentsTable';

describe('InvestmentsTable rendering', () => {
  const mockInvestment = [{
    _id: 1,
    type: 'fixa',
    value: 'R$100,00',
    date: '2019-09-01'
  }];
  const mockDelete = jest.fn()
  const component = shallow(<InvestmentsTable investmentInfo={mockInvestment} deleteInvestment={mockDelete} />); 

  it('should render', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render the table', () => {
    expect(component.find('table').length).toEqual(1);
  });

  it('should render investment date', () => {
    expect(component.find('tr').last().find('td').at(0).text()).toEqual('2019-09-01');
  });

  it('should render investment value', () => {
    expect(component.find('tr').last().find('td').at(1).text()).toEqual('R$100,00');
  });

  it('should render delete icon', () => {
    expect(component.find('.delete').length).toEqual(1);
  });

  it('should call delete function on delete icon click', () => {
    component.find('.delete').simulate('click');
    expect(mockDelete).toHaveBeenCalled();
  });
});