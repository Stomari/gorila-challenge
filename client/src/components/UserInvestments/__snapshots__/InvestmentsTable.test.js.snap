// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvestmentsTable rendering should render 1`] = `ShallowWrapper {}`;
